modin/core/dataframe/pandas/metadata/dtypes.py

```diff
@@ -649,12 +757,14 @@ def concat(cls, values: list) -> "ModinDtypes":
                 raise NotImplementedError(type(val))
 
         try:
-            desc = DtypesDescriptor.concat(preprocessed_vals)
+            desc = DtypesDescriptor.concat(preprocessed_vals, axis=axis)
         except NotImplementedError as e:
             # 'DtypesDescriptor' doesn't support duplicated labels, however, if all values are pandas Serieses,
             # we still can perform concatenation using pure pandas
-            if "duplicated" not in e.args[0].lower() or not all(
-                isinstance(val, pandas.Series) for val in values
+            if (
+                axis == 0
```

Comment:
Why only for this axis?

Reply:
`axis=1` [fails with duplicated labels anyway](https://github.com/modin-project/modin/pull/6759#discussion_r1399677430)

Reply:
A comment in the code would be helpful.

Reply:
done